src/drivers/uavcan/sensors/battery.cpp

```diff
@@ -42,66 +42,129 @@ UavcanBatteryBridge::UavcanBatteryBridge(uavcan::INode &node) :
 	UavcanSensorBridgeBase("uavcan_battery", ORB_ID(battery_status)),
 	ModuleParams(nullptr),
 	_sub_battery(node),
+	_sub_battery_aux(node),
 	_warning(battery_status_s::BATTERY_WARNING_NONE),
 	_last_timestamp(0)
 {
 }
 
 int UavcanBatteryBridge::init()
 {
+	param_get(param_find("UAVCAN_SUB_BAT"), &uavcan_sub_bat);
+
 	int res = _sub_battery.start(BatteryInfoCbBinder(this, &UavcanBatteryBridge::battery_sub_cb));
 
 	if (res < 0) {
 		PX4_ERR("failed to start uavcan sub: %d", res);
 		return res;
 	}
 
+	if (uavcan_sub_bat == 2) {
+		res = _sub_battery_aux.start(BatteryInfoAuxCbBinder(this, &UavcanBatteryBridge::battery_aux_sub_cb));
+
+		if (res < 0) {
+			PX4_ERR("failed to start uavcan sub: %d", res);
+			return res;
+		}
+	}
+
 	return 0;
 }
 
+battery_status_s *
+UavcanBatteryBridge::battery_device_find(uint8_t node_id)
```

Comment:
Given these are private fields of the class could you just do this in place and skip using pointers?